### PR TITLE
Document that UdpSocket::recv and recv_from do not read from the buff…

### PR DESCRIPTION
**Pull request by frehberg**
…er (#657)

### CODE REVIEW

File: src/udp.rs [hunk 1]
```diff
@@ -82,6 +82,13 @@ impl UdpSocket {
 
     /// Receives data from the socket. On success, returns the number of bytes
     /// read and the address from whence the data came.
+    ///
+    /// The function must be called with valid byte array `buf` of sufficient size to
+    /// hold the message bytes. If a message is too long to fit in the supplied buffer,
+    /// excess bytes may be discarded.
+    ///
+    /// The function is overwriting previous content of `buf`. Slicing
+    /// `&buf[..NumRead]` allows efficient access and boundary checking at compile time.
```
Comment (KodrAus) [inline src/udp.rs]: Assuming `NumRead` is the result of `recv_from` I think a more idiomatic name for it is `num_read`, so we know it's some local variable.

File: src/udp.rs [hunk 2]
```diff
@@ -82,6 +82,13 @@ impl UdpSocket {
 
     /// Receives data from the socket. On success, returns the number of bytes
     /// read and the address from whence the data came.
+    ///
+    /// The function must be called with valid byte array `buf` of sufficient size to
```
Comment (KodrAus) [inline src/udp.rs]: I think we still need to mention that this function only writes into the buffer and never reads from it.

File: src/udp.rs [hunk 3]
```diff
@@ -95,7 +102,14 @@ impl UdpSocket {
     }
 
     /// Receives data from the socket previously bound with connect(). On success, returns
-    /// the number of bytes read and the address from whence the data came.
+    /// the number of bytes read.
+    ///
+    /// The function must be called with valid byte array `buf` of sufficient size to
```
Comment (KodrAus) [inline src/udp.rs]: I think we still need to mention that this function only writes into the buffer and never reads from it.

**Comment by frehberg** [comment]
API-Doc has been extended, adding these aspects. I had to re-arrange the text a bit, does it please?

**Comment by carllerche** [comment]
Thanks... this looks good to me. Unfortunately, it looks like Android CI broke... which going to block me from merging until it is fixed 😢

**Comment by carllerche** [comment]
A CI fix has been merged to master. Would you mind rebasing when you have a moment? Thanks!

**Comment by KodrAus** [comment]
@frehberg How's it going? Would you like a hand rebasing this?

**Comment by carllerche** [comment]
Something wonky happened with this PR. The commit was rebased, but didn't make it to the PR. I moved it here: #775

**Pull request closed without merging**